packages/app-project/stores/Project.js

```diff
@@ -0,0 +1,37 @@
+import { flow, getRoot, types } from 'mobx-state-tree'
+import { get } from 'lodash'
+import asyncStates from './asyncStates'
+
+const Project = types
+  .model('Project', {
+    displayName: types.optional(types.string, ''),
+    error: types.optional(types.frozen, null),
+    id: types.optional(types.string, ''),
+    state: types.optional(types.enumeration('state', asyncStates.values), asyncStates.initialized)
+  })
+
+  .volatile(self => ({
+    client: {}
+  }))
+
+  .actions(self => ({
+    afterAttach () {
+      self.client = getRoot(self).client.projects
+    },
+
+    fetch: flow(function * fetch (slug) {
+      self.state = asyncStates.loading
+      try {
+        const project = yield self.client.get({ query: { slug }})
+          .then(response => get(response, 'projects[0]'))
```

Comment:
This is also returning undefined because this is the full response object. The project is in the `body` property. Should we be returning only the body of the response in the client? I didn't do anything with the response because I see having access to `response.ok` or `response.statusCode` would be useful in some async calls.

Reply:
We could just make this line `response => response.body.projects[0]` without the lodash dependency.